src/plugins/intel_gpu/src/plugin/program_builder.cpp

```diff
@@ -318,6 +293,106 @@ ProgramBuilder::ProgramBuilder(cldnn::engine& engine, const ExecutionConfig& con
         m_networkOutputs = *outputs;
 }
 
+ProgramBuilder::ProgramBuilder(const std::shared_ptr<ov::Model>& model, cldnn::engine& engine,
+                                const ExecutionConfig& config,
+                                std::shared_ptr<ov::threading::IStreamsExecutor> task_executor)
+    : m_curBatch(-1)
+    , m_config(config)
+    , m_engine(engine)
+    , queryMode(false)
+    , m_task_executor(task_executor) {
+    if (m_task_executor == nullptr)
+        m_task_executor = cldnn::program::make_task_executor(m_config);
+
+    std::shared_ptr<ov::Model> ngraph_function(model);
+    // Check ngraph function pointer
+    if (!ngraph_function) {
+        OPENVINO_THROW("ov::Model pointer is nullptr");
+    }
+
+    // Set input / output for Parameter and Result in cldnn
+    InferenceEngine::InputsDataMap networkInputs = {};
+    InferenceEngine::OutputsDataMap networkOutputs = {};
+
+    auto create_data_for_result = [&](const ngraph::Output<ngraph::Node>& output,
+                                        const std::string& outName,
+                                        InferenceEngine::DataPtr& ptr) {
+        auto shape = output.get_partial_shape();
+        SizeVector dims(1, 0);
+        if (shape.rank().is_static()) {
+            dims.resize(shape.size(), 0);
+            for (size_t i = 0; i < shape.size(); ++i) {
+                if (shape[i].get_max_length() != -1)  // dimension has an estimation
+                    dims[i] = shape[i].get_max_length();
+            }
+        }
+        auto rank = shape.rank().is_static() ? shape.rank().get_length() : -1;
+        const auto rankLayout = rank < 0 ? InferenceEngine::Layout::BLOCKED : TensorDesc::getLayoutByRank(rank);
+        const auto precision = InferenceEngine::details::convertPrecision(output.get_element_type());
+        ptr.reset(new InferenceEngine::Data(outName, {precision, dims, rankLayout}));
```

Comment:
can't we create it as shared_ptr?

Reply:
Also I think we can add assert that rank should be static because we dont support dynamic rank